### PR TITLE
feat: `deepsource analyzer init` command

### PR DESCRIPTION
**Pull request by siddhant-deepsource**
Signed-off-by: Siddhant N Trivedi [siddhant@deepsource.io](mailto:siddhant@deepsource.io)

### CODE REVIEW

**Comment by siddhant-deepsource** [comment]
Oops..Please don't review as of now. Will add more tests.

File: command/analyzer/initialize/init.go [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+	"strings"
+
+	validator "github.com/deepsourcelabs/cli/analyzers/validator"
+	"github.com/deepsourcelabs/cli/utils"
+	"github.com/pelletier/go-toml"
+	"github.com/pterm/pterm"
+	"github.com/spf13/cobra"
+)
+
+var SDKLanguageOpts = []string{"Python", "Go", "JavaScript"}
```
Comment (sourya-deepsource) [inline command/analyzer/initialize/init.go:18]: Take this value from a function. If there are no SDKs, the function should return an empty array. The function should take the data from a GitHub repo which will act as an SDK registry, but that can be implemented later.

File: command/analyzer/initialize/init.go [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+
+type SDKResponse struct {
+	SDKRequired bool
+	SDKLanguage string
+}
+
```
Comment (sourya-deepsource) [inline command/analyzer/initialize/init.go:24]:
```suggestion
type IAnalyzerInitOpts interface {
	SingleLineInputPrompt func(string, string, string) (string, error)
	DescriptionPrompt     func(string, string, string) (string, error)
	ConfirmationPrompt    func(string, string) (bool, error)
	SingleOptionPrompt    func(string, string, []string) (string, error)
}
```

Comment (sourya-deepsource) [reply command/analyzer/initialize/init.go:24]: You can add this interface and ensure that the struct used in tests satisfy this interface.

File: command/analyzer/initialize/init.go [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+	// Eg: @deepsource/armory -> Armory
+	defaultAnalyzerName := strings.Title(strings.SplitAfter(a.AnalyzerTOMLData.Shortcode, "/")[1])
+
+	// Collect name of the Analyzer
+	msg = "Please enter the name of the Analyzer?"
+	helpText = "The name of the Analyzer."
+	a.AnalyzerTOMLData.Name, err = a.SingleLineInputPrompt(msg, helpText, defaultAnalyzerName)
+	if err != nil {
+		return nil, err
+	}
```
Comment (sourya-deepsource) [inline command/analyzer/initialize/init.go:91]:
```suggestion
	if a.AnalyzerTOMLData.Name, err = a.SingleLineInputPrompt(msg, helpText, defaultAnalyzerName); err != nil {
		return nil, err
	}
```

File: command/analyzer/initialize/init.go [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+	helpText = "What does the Analyzer do?"
+	a.AnalyzerTOMLData.Description, err = a.DescriptionPrompt(msg, helpText, "")
+	if err != nil {
+		return nil, err
+	}
+
+	// Check if DeepSource SDK is needed or not?
+	msg = "Would you like to use DeepSource Analyzer SDK to build your Analyzer?"
+	helpText = "DeepSource SDKs help you to easily create an Analyzer"
+	a.SDKInput.SDKRequired, err = a.ConfirmationPrompt(msg, helpText)
```
Comment (sourya-deepsource) [inline command/analyzer/initialize/init.go:104]: Only show this option when the SDKLanguageOpts is populated.

File: utils/config.go [hunk 5]
```diff
@@ -18,3 +22,18 @@ func ExtractProjectRootPath() (string, error) {
 	// Removing trailing null characters
 	return strings.TrimRight(string(output), "\000\n"), nil
 }
+
+func InitAnalyzerConfigurationPaths() (string, string, string) {
```
Comment (sourya-deepsource) [inline utils/config.go:26]: This should be inside the `analyzer` package.

**Review by sourya-deepsource — CHANGES_REQUESTED** [review]

**Review by sourya-deepsource — APPROVED** [review]

**Pull request merged**